scipy/stats/_continuous_distns.py

```diff
@@ -757,37 +757,103 @@ class burr_gen(rv_continuous):
 
     .. math::
 
-        f(x, c, d) = c d x^{-c-1} (1+x^{-c})^{-d-1}
+        f(x, c, d) = c d x^{-c - 1} / (1 + x^{-c})^{d + 1}
 
     for :math:`x >= 0` and :math:`c, d > 0`.
 
     `burr` takes :math:`c` and :math:`d` as shape parameters.
 
     This is the PDF corresponding to the third CDF given in Burr's list;
-    specifically, it is equation (11) in Burr's paper [1]_.
+    specifically, it is equation (11) in Burr's paper [1]_. The distribution
+    is also commonly referred to as the Dagum distribution [2]_. If the
+    parameter :math:`c < 1` then the mean of the distribution does not
+    exist and if :math:`c < 2` the variance does not exist [2]_.
 
     %(after_notes)s
 
     References
     ----------
     .. [1] Burr, I. W. "Cumulative frequency functions", Annals of
        Mathematical Statistics, 13(2), pp 215-232 (1942).
+    .. [2] https://en.wikipedia.org/wiki/Dagum_distribution
+    .. [3] Kleiber, Christian. "A guide to the Dagum distributions."
+       Modeling Income Distributions and Lorenz Curves  pp 97-117 (2008).
 
     %(example)s
 
     """
-    _support_mask = rv_continuous._open_support_mask
+    # Do not set _support_mask to rv_continuous._open_support_mask
+    # Whether the left-hand endpoint is suitable for pdf evaluation is dependent
+    # on the values of c and d
 
     def _pdf(self, x, c, d):
         # burr.pdf(x, c, d) = c * d * x**(-c-1) * (1+x**(-c))**(-d-1)
-        return c * d * (x**(-c - 1.0)) * ((1 + x**(-c))**(-d - 1.0))
+        xx = np.asarray(x)
+        dtyp = np.find_common_type([xx.dtype, np.float64], [])
+        output = np.zeros(np.shape(xx), dtyp)
+        cond = (xx == 0)
+        if np.any(cond):
+            xz = xx[cond]
+            output[cond] = c * d * (xz**(c*d-1)) / (1 + xz**c)
+        if np.any(~cond):
+            xnz = xx[~cond]
+            output[~cond] = c * d * (xnz ** (-c - 1.0)) / ((1 + xnz ** (-c)) ** (d + 1.0))
+        if output.ndim == 0:
+            return output[()]
+        return output
+
+    def _logpdf(self, x, c, d):
```

Comment:
Not sure about the "numerical precision". Why is the `_logpdf` different from/better than the `_pdf`?

Reply:
with `_logpdf` you are working in "log space" so you retain more bits of precision and only move out of log-space and back into the `_pdf` space at the end. I'm not sure if it will practically make much difference in this case for most values of `c` and `d` but for some values it might matter, plus it would remove the redundant code that is shared across `_pdf` and `_logpdf`

Reply:
Can you explain further why working in "log space" retains more precision in this case?

Reply:
I don't see why `sc.xlog1py` would be always more precise.  For some values of `x, c, d`, `(1 + x ** (-c)) ** (d + 1.0)`can be done exactly whereas computing with logs doesn't guarantee preserving exactness.  Additionally `z -> exp(z)` has a relative condition number of `|z|`, so any extra precision due to computing in log-space can be lost when converting back to probability space.  An implementation of `a**b` might choose to do the operation in log-space, but it is not required to.